documentation/correctness.adoc

```diff
@@ -0,0 +1,133 @@
+Violation is to be reported in terms of the ID of the contain**ing** serialized node, the ID of the contain**ed** serialized node, and the `parent` ID stated by that.
+. No duplicate containments, i.e. nodes are not the children of multiple other nodes, nor of themselves.
+Violation is to be reported in terms of the ID of the duplicately-contained serialized node, and the IDs of all serialized nodes that claim to contain it, including the links by which this happens, as well the index in the value of those links.
+. No containment cycles, i.e. nodes are not containing themselves.
+. **TODO**  extend for annotations
+
+At this point, links (so: containments and references) don't need to resolve (yet).
+Concretely, this means that IDs that are members of the `children` (for containments) or `targets` (for references) arrays, don't need to resolve to a node in the same JSON serialization.
+They could -- or even: should -- reside in another serialization.
+That pertains specifically to references to language entities of the LIonCore builtins language, which is assumed to be imported/available implicitly.
+If all links do resolve, we say the serialization chunk is _self-contained_.
+
+=== Meta-referential
+
+All the metamodels specified in the serialization chunk should be _made available_ to the client('s deserializer) in the form of a runtime representation that the deserializer can reconstruct from the serializatino.
+**Note** that this is a condition that pertains to the deserializing client, rather than on the serialization chunk.
+Incorrectness could be partially recovered from by partially deserializing into "`reflective`" nodes.
+
+=== Meta-structural
+
+The seralization matches the metamodels - in particular:
+
+* Concept identifiers (tuples with a `key` value) are uniquely resolvable to a concept in a metamodel.
+Incorrectness is to be reported in terms of the ID of the serialized and the key of the concept which can not be resolved.
+* For every object corresponding to a serialized node:
+** For every entry of the `properties`, `children`, or `references` arrays, the `key` in the `property`, `containment`, resp. `reference` sub object is the key of a feature of the concept pointed to by the `concept` identifier.
+Incorrectness is to be reported in terms of the ID of the serialized node, the feature type (property/containment/reference), the index of the offending entry, and the key which can not be resolved.
+** For every entry of a `properties` array, the value of the `value` property conforms to the `type` specified by the `Property` feature corresponding to the key.
+
+Incorrectness is to be reported in terms of the ID of the serialized node, the feature type (property/containment/reference), the index of the offending entry, and the key which can not be resolved.
+"`To resolve a key`" means that we can find a concept or feature in some metamodel with that key, and with (exactly-)matching version.
```

Comment:
I think there's a formatting issue, this doesn't belong to property values

Reply:
I don't see the formatting issue.

Reply:
Me neither (-:

Maybe I stumbled on "Incorrectness is to be reported  ..." and '"`To resolve a key`" means ...' to be rendered in the same paragraph.